pkgs/applications/emulators/wibo/default.nix

```diff
@@ -1,24 +1,40 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchurl
 , cmake
+, unzip
 }:
 
 stdenv.mkDerivation rec {
   pname = "wibo";
-  version = "0.2.0";
+  version = "0.2.4";
 
   src = fetchFromGitHub {
     owner = "decompals";
-    repo = "WiBo";
+    repo = "wibo";
     rev = version;
-    sha256 = "sha256-zv+FiordPo7aho3RJqDEe/1sJtjVt6Vy665VeNul/Kw=";
+    hash = "sha256-dpfKSiIWE9L5BLPH2t8RsUz7Ufkdo/5zn1dewaEgJl0=";
   };
 
   nativeBuildInputs = [
     cmake
+    unzip
   ];
 
+  doCheck = true;
+  # Test step from https://github.com/decompals/wibo/blob/main/.github/workflows/ci.yml
+  checkPhase = let
+    gc = fetchurl {
+      url = "https://cdn.discordapp.com/attachments/727918646525165659/917185027656286218/GC_WII_COMPILERS.zip";
```

Comment:
These are unfree and should probably not be fetched on hydra, although the testing is helpful.

Reply:
Maybe disable check by default?

Reply:
Sounds good. Maybe also wrap this fetchurl in a srcOnly with license = unfree just to be sure.

Reply:
In such case `allowUnfree = true` will be required, despite the fact that `doCheck = false;`. I don't think it's good, since wibo is actually free (MIT).

Reply:
Right, we would also have to wrap the checkPhase in an `optionalString doCheck` so that gc would not evaluate when allowUnfree = false.

```diff
   doCheck = false;
   # Test step from https://github.com/decompals/wibo/blob/main/.github/workflows/ci.yml
   checkPhase = let
-    gc = fetchurl {
-      url = "https://cdn.discordapp.com/attachments/727918646525165659/917185027656286218/GC_WII_COMPILERS.zip";
-      hash = "sha256-iT5i/PUe37aEB9zoW0k+ic9ZDnkWn6Us6IOHtsjI9qc=";
+    gc = srcOnly {
+      name = "GC_WII_COMPILERS";
+      src = fetchurl {
+        url = "https://cdn.discordapp.com/attachments/727918646525165659/917185027656286218/GC_WII_COMPILERS.zip";
+        hash = "sha256-iT5i/PUe37aEB9zoW0k+ic9ZDnkWn6Us6IOHtsjI9qc=";
+      };
+      meta.license = lib.licenses.unfree;
     };
-  in ''
+  in lib.optionalString doCheck ''
     unzip ${gc}
     MWCIncludes=. ./wibo GC/2.7/mwcceppc.exe -c ../test/test.c
     file test.o | grep "ELF 32-bit"
```

I think this is important because Hydra should not under any circumstance download and serve GC_WII_COMPILERS.

Reply:
Oh, and I think you'll need a couple more changes for this to evaluate now, because srcOnly doesn't like using a .zip file with fetchurl. This works for me:

```diff
 { lib
 , stdenv
 , fetchFromGitHub
-, fetchurl
+, fetchzip
 , srcOnly
 , cmake
 , unzip
@@ -28,15 +28,15 @@ stdenv.mkDerivation rec {
   checkPhase = let
     gc = srcOnly {
       name = "GC_WII_COMPILERS";
-      src = fetchurl {
+      src = fetchzip {
         url = "https://cdn.discordapp.com/attachments/727918646525165659/917185027656286218/GC_WII_COMPILERS.zip";
-        hash = "sha256-iT5i/PUe37aEB9zoW0k+ic9ZDnkWn6Us6IOHtsjI9qc=";
+        hash = "sha256-o+UrmIbCsa74LxtLofT0DKrTRgT0qDK5/V7GsG2Zprc=";
+        stripRoot = false;
       };
       meta.license = lib.licenses.unfree;
     };
   in lib.optionalString doCheck ''
-    unzip ${gc}
-    MWCIncludes=. ./wibo GC/2.7/mwcceppc.exe -c ../test/test.c
+    MWCIncludes=. ./wibo ${gc}/GC/2.7/mwcceppc.exe -c ../test/test.c
     file test.o | grep "ELF 32-bit"
   '';
 
```